npm-package/src/controllers/otelController.ts

```diff
@@ -1,25 +1,95 @@
+type MainController = (req: Request, res: Response, next: NextFunction) => void
+type HelperController = (req: Request) => {}
 
-    },
-    parseNodeRequest : (req: Request, res: Response, next: NextFunction) => {
+export interface OtelControllerType {
+  parseAllRequest: MainController
+  parseNodeRequest: HelperController
+  parseNextJSRequest: HelperController
+}
 
-    } 
-};
+export const otelController: OtelControllerType = {
+  parseAllRequest: (req: Request, res: Response, next: NextFunction) => {
+    //need to diffrentiate between nextJS and Node application since parsing of the data is different
+    let data = {}
+
+    if (
+      req.body.resourceSpans[0].resource.attributes[4].value.stringValue ===
+      'next.js'
+    ) {
+      data = otelController.parseNextJSRequest(req)
+    }
+
+    console.log(data)
+
+    return next()
+  },
+  parseNextJSRequest: (req: Request) => {
+    const data: any = {}
+
+    if (req.body.resourceSpans[0].scopeSpans[0].spans[0]?.traceId) {
+      data.traceId = req.body.resourceSpans[0].scopeSpans[0].spans[0]?.traceId
+    }
+
+    if (req.body.resourceSpans[0].scopeSpans[0].spans[0]?.spanId) {
+      data.spanId = req.body.resourceSpans[0].scopeSpans[0].spans[0]?.spanId
+    }
+
+    if (
+      req.body.resourceSpans[0].resource.attributes[4].value.stringValue ===
+      'next.js'
+    ) {
+      const applicationType =
+        req.body.resourceSpans[0].resource.attributes[4].value.stringValue
+      data.ApplicationType = applicationType
+    }
+
+    if (req.body.resourceSpans[0].resource?.attributes[0]?.value?.stringValue) {
+      data.originatingService =
+        req.body.resourceSpans[0].resource?.attributes[0]?.value?.stringValue
+    }
+
+    if (req.body.resourceSpans[0]?.scopeSpans[0]?.spans[0]?.name) {
+      data.method = req.body.resourceSpans[0]?.scopeSpans[0]?.spans[0]?.name
+    }
+
+    let currentSpan =
+      req.body.resourceSpans[0].scopeSpans[0].spans[0].attributes
+    for (let i = 0; i < currentSpan.length; i++) {
+      if (currentSpan[i].key === 'http.status_code')
+        data.status = currentSpan[i].value.intValue
+    }
+
+    let httpArray = req.body.resourceSpans[0].scopeSpans[0].spans[0].attributes
+    for (let i = 0; i < httpArray.length; i++) {
+      if (httpArray[i].key === 'http.flavor')
+        data.protocol = httpArray[i].value.stringValue
+    }
+
+    data.startTime =
+      req.body.resourceSpans[0].scopeSpans[0].spans[0].startTimeUnixNano
+    data.endTime =
+      req.body.resourceSpans[0].scopeSpans[0].spans[0].endTimeUnixNano
+
+    // console.log('application type', req.body.resourceSpans[0].scopeSpans[0].spans[0].attributes);
+
+    for (let i = 0; i < currentSpan.length; i++) {
+      if (currentSpan[i].key === 'size') {
+        data.size = currentSpan[i].value.intValue
+      }
+    }
+
+    return data
+  },
+  parseNodeRequest: (req: Request) => {
+    const data = {}
+
+    return data
```

Comment:
Do we have the functionality of parsing node requests? it seems this just returns an empty object

Reply:
this will be worked on right now